openlibrary/plugins/openlibrary/partials.py

```diff
@@ -0,0 +1,206 @@
+import json
+from urllib.parse import parse_qs
+
+import web
+
+from infogami.utils import delegate
+from infogami.utils.view import render_template
+from openlibrary.core import cache
+from openlibrary.core.fulltext import fulltext_search
+from openlibrary.core.lending import get_available
+from openlibrary.plugins.worksearch.code import do_search, work_search
+from openlibrary.plugins.worksearch.subjects import get_subject
+from openlibrary.utils import dateutil
+from openlibrary.views.loanstats import get_trending_books
+
+
+def _get_relatedcarousels_component(workid):
+    if 'env' not in web.ctx:
+        delegate.fakeload()
+    work = web.ctx.site.get('/works/%s' % workid) or {}
+    component = render_template('books/RelatedWorksCarousel', work)
+    return {0: str(component)}
+
+
+def get_cached_relatedcarousels_component(*args, **kwargs):
+    memoized_get_component_metadata = cache.memcache_memoize(
+        _get_relatedcarousels_component,
+        "book.bookspage.component.relatedcarousels",
+        timeout=dateutil.HALF_DAY_SECS,
+    )
+    return (
+        memoized_get_component_metadata(*args, **kwargs)
+        or memoized_get_component_metadata.update(*args, **kwargs)[0]
+    )
+
+
+class CarouselCardPartial:
+    def __init__(self):
+        self.i = web.input(params=None)
+
+    def generate(self) -> dict:
+        # Determine query type
+        params = self.i or {}
+        query_type = params.get("queryType", "")
+
+        # Do search
+        search_results = self._make_book_query(query_type, params)
+
+        # Render cards
+        cards = []
+        layout = params.get("layout")
+        key = params.get("key") or ""
+        for index, book in enumerate(search_results):
+            lazy = index > 5
```

Comment:
At most, I think we show 5 images in the carousel (no matter the breakpoints) so perhaps if there are > 5 (magic number), we lazy load everything else?

If so, we may want to call this MAX_VISIBLE_CARDS = 5